packages/react-query/src/QueryClientProvider.tsx

```diff
@@ -55,23 +31,16 @@ type QueryClientProviderPropsBase = {
   client: QueryClient
   children?: React.ReactNode
 }
-type QueryClientProviderPropsWithContext = ContextOptions & {
-  contextSharing?: never
-} & QueryClientProviderPropsBase
-type QueryClientProviderPropsWithContextSharing = {
-  context?: never
-  contextSharing?: boolean
-} & QueryClientProviderPropsBase
 
-export type QueryClientProviderProps =
-  | QueryClientProviderPropsWithContext
-  | QueryClientProviderPropsWithContextSharing
+type QueryClientProviderPropsWithContext = ContextOptions &
+  QueryClientProviderPropsBase
+
+export type QueryClientProviderProps = QueryClientProviderPropsWithContext
```

Comment:
Would it make sense to simply define the `QueryClientProviderProps` like
```suggestion
export type QueryClientProviderProps = {
  client: QueryClient
  children?: React.ReactNode
} & ContextOptions
```

Reply:
Yeah this makes sense, I'll do it for the solid-query implementation too